### PR TITLE
feat(argo-cd): Create optional annotations and labels for redis service

### PR DESCRIPTION
**Pull request by arunanandhan**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/redis/service.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ metadata:
   name: {{ template "argo-cd.redis.fullname" . }}
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.redis.name "name" .Values.redis.name) | nindent 4 }}
+  {{- with .Values.redis.serviceAnnotations }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/service.yaml:9]: Please use `.Values.redis.service.annotations` as other components implement it this way.

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Review by jwenz723 — APPROVED** [review]

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: 2.1.2
 description: A Helm chart for ArgoCD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 3.22.1
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:5]: L24 needs to be replaced with an appropriate description

**Review by mkilchhofer — APPROVED** [review]
lgtm. I moved some parts around and also implemented this feature for additional labels like other components support.

File: charts/argo-cd/templates/redis/service.yaml [hunk 3]
```diff
@@ -6,10 +6,17 @@ metadata:
...
+    {{- end }}
+  {{- with .Values.redis.service.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
   ports:
   - port: {{ .Values.redis.servicePort }}
-    targetPort: {{ .Values.redis.servicePort }}
+    targetPort: {{ .Values.redis.containerPort }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/service.yaml:19]: Oh and while reviewing, I saw that this needs to be fixed

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**